docs/rules/component-name-unique.md

```diff
@@ -0,0 +1,113 @@
+unique through the whole spec.
+
+- schema: The code generator creates a class for each schema.
+  If they are not uniquely named, the generator appends numbers. These numbers are non-deterministic.
+  By adding a new schema with the same component name it could change the name (appended number) of another one.
+- parameter: The code generator creates a class for each parameter.
+  If they are not uniquely named, the generator appends numbers. These numbers are non-deterministic.
+  By adding a new parameter with the same component name it could change the name (appended number) of another one.
+- response: The code generator tends to reuse the first one and drops the other ones with the same component name.
+- requestBody: The code generator tends to reuse the first one and drops the other ones with the same component name.
+
+This clearly is not optimal. Having unique component names prevents these problems.
+
+## Configuration
+
+| Option        |Type| Description                                                                              |
+|---------------|---|------------------------------------------------------------------------------------------|
+| severity      |string| Possible values: `off`, `warn`, `error`. Default `off` (in `recommended` configuration). |
+| schemas       |string| Possible values: `off`, `warn`, `error`. Default: not set. |
+| parameters    |string| Possible values: `off`, `warn`, `error`. Default: not set. |
+| responses     |string| Possible values: `off`, `warn`, `error`. Default: not set. |
+| requestBodies |string| Possible values: `off`, `warn`, `error`. Default: not set. |
+
+An example configuration:
+
+```yaml
+rules:
+  component-name-unique:
+    schemas: error
+    parameters: off
+    responses: warn
+    requestBodies: warn
+```
+
+## Examples
+
+
+Given this configuration:
+
+```yaml
+rules:
+  component-name-unique: error
+```
+
+### Example of **incorrect** schema files
+
+file1.yaml:
+```yaml
+components:
+  schemas:
+    FooSchema:
+      type: object
+      properties:
+        field:
+          $ref: './file2.yaml#/components/schemas/FooSchema'
+```
+
+file2.yaml:
+```yaml
+components:
+  schemas:
+    FooSchema:
+      type: object
+      properties:
+        otherField:
+          type: string
+```
+
+### Example of a **correct** schema files
+
+file1.yaml:
+```yaml
+components:
+  schemas:
+    FooSchema:
+      type: object
+      properties:
+        field:
+          $ref: './file2.yaml#/components/schemas/AnotherFooSchema'
+```
+
+file2.yaml:
+```yaml
+components:
+  schemas:
+    AnotherFooSchema:
+      type: object
+      properties:
+        otherField:
+          type: string
+```
+
+## Relates rules
+
+- [no-unused-components](./no-unused-components.md)
+
+## Resources
+
+- [Rule source](https://github.com/Redocly/redocly-cli/blob/main/packages/core/src/rules/oas3/component-name-unique.ts)
```

Comment:
@tatomyr the `lintcheck` doesn't like this due to the link not yet existing [[link to failure](https://github.com/Redocly/redocly-cli/actions/runs/5347202202)]. But it will exist once this PR gets merged. How was this done for the other rules that link to their own file?
Only other ways I see:
1. Link it to latest commit in the PR (not so nice, but nicer than not having it)
2. Remove the link (not that nice, as it is good to have it same as the others)

Okay with whatever is the preferred way from you.

For now I will remove it just so that the next run will pass...

Reply:
Don't worry, we'll add the link ourselves! 

Just in case, we usually create two separate merge requests - one for the rule itself, and one for the docs (I've recently updated our [Contributing guide](https://github.com/Redocly/redocly-cli/blob/main/.github/CONTRIBUTING.md#built-in-rules-changes) regarding the flow).